utility/src/main/java/org/oppia/android/util/networking/ProdNetworkConnectionUtil.kt

```diff
@@ -0,0 +1,36 @@
+package org.oppia.android.util.networking
+
+import android.content.Context
+import android.net.ConnectivityManager
+import org.oppia.android.util.networking.NetworkConnectionUtil.ConnectionStatus
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * [NetworkConnectionUtil] which gets the current [ConnectionStatus] of the device in production builds.
+ */
+@Singleton
+class ProdNetworkConnectionUtil @Inject constructor(
+  private val context: Context
+) : NetworkConnectionUtil {
+
+  override fun getCurrentConnectionStatus(): ConnectionStatus {
+    val connectivityManager = context.getSystemService(
+      Context.CONNECTIVITY_SERVICE
+    ) as ConnectivityManager
+    return connectivityManager.activeNetworkInfo?.let { activeNetwork ->
```

Comment:
Added permissions to the Manifest file. 

Regarding the deprecated version, it was already in use in the codebase so I just used this only and I think there might be some reason for it. What I think is that our targeted version is API 28 and `activeNetworkInfo` is deprecated in API 29 and above. So, it may be a reason that we are still using it. Also, there is an alternative to this ([reference](https://stackoverflow.com/a/57284789)). But, it works only on API levels greater than 23 and our minimum API level is 19. So, even in this case, we won't be able to cover all the API levels. WDYT should we do here?

Reply:
Looks like there is no backward compatibility here. We have to go with this only. If I come up with something, I will create a separate issue for it as other than this PR there are places where we are using this.

Reply:
According to https://developer.android.com/reference/android/net/ConnectivityManager#getActiveNetworkInfo() this should be available from API 1. Is that not the case?

Also, I suspect the "correct" way to do this is to have a TargetApi method for 28 that uses the old version, and a 29+ method for the new version and use an if-statement to choose. It's probably worth filing & adding a TODO here to update this code to support both old and new APIs to make sure we're keeping up with SDK changes.

Reply:
Filed an issue (#3616) and added the corresponding TODO. PTAL.